### DISCUSSION TITLE
Problem with Field Array Validation in Form

### DISCUSSION THREAD

**Discussion by yldshv** [data-driven-forms/react-forms#1445] [Need help] [answered]
Hello everyone, it's me again,

This time I have a new problem for which I can't find a solution. I hope you can help me as always. It might also be a bug.

When I have a form and use a field array within the form, with the fields in the array being required, adding a new item to the array does not consider the overall validity of the form. 

For example, if I have filled out all fields in my form and it is valid, and then I add another item to the field array list (which contains required fields that are now empty), the form still remains valid overall.

Does anyone have any ideas or solutions for this?

Best regards,
Daniel

**Comment by rvsia** [comment]
Hello, 

would it be possible to create a live example (code sandbox, stackblitz) to replicate the issue? 😊 

Thanks

**Comment by yldshv** [comment]
Hi @rvsia ,

I tried to recreate the whole thing as closely as possible in a sandbox. If you look into your console, you should see what I mean.

[CodeSandBox](https://codesandbox.io/p/sandbox/kind-shape-nps7fn?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clpmarfgs00072v6knymdemwq%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clpmarfgr00032v6kdu026iwt%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clpmarfgr00042v6kzj227398%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clpmarfgr00062v6kel3us441%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clpmarfgr00032v6kdu026iwt%2522%253A%257B%2522id%2522%253A%2522clpmarfgr00032v6kdu026iwt%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpmb0f6400022v6kqxmwo1iy%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A98%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A98%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clpmb0f6400022v6kqxmwo1iy%2522%257D%252C%2522clpmarfgr00062v6kel3us441%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpmarfgr00052v6khh87yfsc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clpmarfgr00062v6kel3us441%2522%252C%2522activeTabId%2522%253A%2522clpmarfgr00052v6khh87yfsc%2522%257D%252C%2522clpmarfgr00042v6kzj227398%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clpmarfgr00042v6kzj227398%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Reply by rvsia** [reply, marked as answer]
Thank you for providing this example.

I think there is an issue using the `useEffect`.

If you do only this, it works fine:

```jsx
  const FieldListener = () => {
    const { valid } = useFormApi();

    console.log("valid", valid);

    return null;
  };
```

Also the `submit` is working correctly.

**Comment by yldshv** [comment]
@rvsia thank you. That helped tbh idk why we added that useEffect there xD
